src/forms/forms.service.ts

```diff
@@ -35,31 +39,102 @@ export const formSelect = {
                     },
                 },
             },
+            subQuestions: {
+                select: {
+                    id: true,
+                    order: true,
+                    inputType: {
+                        select: {
+                            id: true,
+                            name: true,
+                        },
+                    },
+                    text: true,
+                    description: true,
+                    answerRequired: true,
+                    multipleAllowed: true,
+                    optionGroup: {
+                        select: {
+                            optionChoices: {
+                                select: {
+                                    id: true,
+                                    text: true,
+                                },
+                            },
+                        },
+                    },
+                    createdAt: true,
+                    updatedAt: true,
+                },
+            },
+            createdAt: true,
+            updatedAt: true,
         },
     },
-};
+} as Prisma.FormSelect;
 
 @Injectable()
 export class FormsService {
     constructor(private prisma: PrismaService) {}
 
-    getAllForms() {
-        return this.prisma.form.findMany({
+    async getAllForms() {
+        const data = await this.prisma.form.findMany({
             select: formSelect,
         });
+
+        const subQuestionsIds = [];
+
+        data.forEach((form) => {
+            form.questions.forEach((question) => {
+                const currentQuestion = question as Question & {
+                    subQuestions: Question[];
+                };
+                currentQuestion.subQuestions.forEach((subQuestion) => {
+                    subQuestionsIds.push(subQuestion.id);
+                });
+            });
+        });
+
+        for (let i = 0; i < data.length; i++) {
+            const form = data[i];
+            const filteredQuestions = form.questions.filter(
+                (i) => !subQuestionsIds.includes(i.id),
+            );
+
+            form.questions = filteredQuestions;
+        }
+
+        return data;
     }
 
     async getFormById(formId: number) {
-        const form = await this.prisma.form.findUnique({
+        const data = await this.prisma.form.findUnique({
```

Comment:
i think this should be `form` as the returned object is a form object
you can see an example in the prisma docs